Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -13,18 +13,145 @@ static void Main(string[] args)
+                    minTime = totalSeconds + 0;
+                    maxTime = minTime + 59;
+                }else if(showTime.Contains("minutes")){
+                    string[] temp = showTime.Split(' ');
+                    int tempMin = Int32.Parse(temp[0]);
+                    minTime = totalSeconds + 60*tempMin + 0;
+                    maxTime = minTime + 59;
+                }else if(showTime.Contains("hours")){
+                    string[] temp = showTime.Split(' ');
+                    int tempHr = Int32.Parse(temp[0]);
+                    minTime = totalSeconds + 3660*tempHr ;
+                    maxTime =  minTime + 59*60 + 59;
+                }
+                // But a day can have only 86400 seconds, so we need to take mod for minTime and maxTime
+                minTime = minTime % 86400;
+                maxTime = maxTime % 86400;
+            }
+
+        }
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
             // Add your code here.
-            throw new NotImplementedException();
+            int minmaxTime,maxminTime;
+            int hh,mm,ss; // resultant hour, minutes and seconds
+            string result = ""; // result string to be returned
+
+            //Taking input and processing that input:
+            int len = exactPostTime.Length;
+            TimeClass[] TT = new TimeClass[len]; // TT stands for TestCase Time !
+            for (int i = 0; i < len; i++)
+            {
+                TT[i] = new TimeClass(exactPostTime[i], showPostTime[i]);
+                TT[i].Process();   
+            }
+            
+            // Now lets find the overlapping time cases
+            maxminTime = TT[0].minTime;
+            for (int i = 0; i < len; i++)
+            {
+                if(TT[i].minTime > maxminTime)
+                    maxminTime= TT[i].minTime;
+            }
+            minmaxTime = TT[0].maxTime;
+            for (int i = 0; i < len; i++)
+            {
+                if(TT[i].maxTime < minmaxTime)
+                    minmaxTime= TT[i].maxTime;
+            }
+
+            
+            // Now we already have the boundary conditions.
+            if(maxminTime < minmaxTime){
+                
+                // we have a result to be returned in this case!
+                // result to be considered is - maxmintime(seconds)
+                // so, now we will convert maxmintime to hh:mm:ss format
+                hh = maxminTime / 3600;
+                hh = hh % 24;
+                maxminTime = maxminTime % 3600;
+                mm = maxminTime / 60;
+                maxminTime = maxminTime % 60;
+                ss = maxminTime;
+
+
+                // Returning result in format HH:MM:SS, we need to convert single digit(9:7:5) numbers to double digit(09:07:05)
+                // adding hours to result
+                if(hh>9){
+                     result = result + $"{hh}:";

```

Comment:
Try to use `string.Format` to manipulate `string`. String is immutable hence every time you modify it, a new string is created. This will eat CPU and Memory under heavy load. Where string manipulation is required try to use 'StringBuilder'.